### PR TITLE
Normal incidence

### PR DESCRIPTION
**Pull request by chchatte92**
### Briefly, what does this PR introduce?
The PR tries to estimate the incidence angle of the impinging MC photons.
For the testing purpose I am just using the existic test_pixel_gap_cuts.cpp; will move to a total new code in case the procedure is right.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators
@c-dilks is aware of the studies and he has been asked to kindly review the changes made in EICRecon to get the sensor normalZ.

### CODE REVIEW

File: src/test_pixel_gap_cuts.cpp [hunk 1]
```diff
@@ -72,6 +74,12 @@ int main(int argc, char** argv) {
 
         auto cellID = sim_hit.getCellID();
         auto pos    = sim_hit.getPosition();
+        auto mom    = sim_hit.getMomentum();
+        TVector3 p; p.SetX(mom.x); p.SetY(mom.y); p.SetZ(mom.z);
```
Comment (c-dilks) [inline src/test_pixel_gap_cuts.cpp:78]:
```suggestion
        TVector3 p(mom.x, mom.y, mom.z);
```

File: src/test_pixel_gap_cuts.cpp [hunk 2]
```diff
@@ -72,6 +74,12 @@ int main(int argc, char** argv) {
 
         auto cellID = sim_hit.getCellID();
         auto pos    = sim_hit.getPosition();
+        auto mom    = sim_hit.getMomentum();
+        TVector3 p; p.SetX(mom.x); p.SetY(mom.y); p.SetZ(mom.z);     
+        auto normZ  = geo.GetSensorSurface(cellID);
+    
+        double angle = normZ.Dot(p.Unit());
```
Comment (c-dilks) [inline src/test_pixel_gap_cuts.cpp:81]: Do you need $\arccos$ here?

Are you sure $|\mbox{normZ}|=1$?

Comment (chchatte92) [reply src/test_pixel_gap_cuts.cpp:81]: > Do you need arccos here?
Damn! That's was the case. TVector3 It returns Cos(angle).   
> Are you sure |normZ|=1?
No. In my local copy I changed it to normZ.Unit().

Comment (chchatte92) [reply src/test_pixel_gap_cuts.cpp:81]: @c-dilks, This distribution I like very much!

image

File: src/test_pixel_gap_cuts.cpp (hunk 1 above)
Comment (chchatte92) [reply src/test_pixel_gap_cuts.cpp:78]: The one you suggested was what I tried for the first time. The compilation failed and it complained that this method not found (LoL) and so I had to move to SetX(). This morning I have updated my `eic-shell`.

File: src/test_pixel_gap_cuts.cpp (hunk 2 above)
Comment (chchatte92) [reply src/test_pixel_gap_cuts.cpp:81]: I thought that was more meaningful. but, I can fill the histogram before subtracting from 180 degrees.

Comment (c-dilks) [reply src/test_pixel_gap_cuts.cpp:81]: Angle of incidence should be $\in[0,90]^\circ$
https://en.wikipedia.org/wiki/Angle_of_incidence_(optics)

Comment (c-dilks) [reply src/test_pixel_gap_cuts.cpp:81]: Good to see there is a normal incidence peak though, which is what we tried to achieve when tuning the optics by eye.

Comment (chchatte92) [reply src/test_pixel_gap_cuts.cpp:81]: > Good to see there is a normal incidence peak though, which is what we tried to achieve when tuning the optics by eye.

image
We did a good job!

**Comment by c-dilks** [comment]
With https://github.com/eic/EICrecon/pull/1068 merged, is this ready?

**Pull request closed without merging**